services/src/main/java/org/keycloak/crypto/ServerMacSignatureVerifierContext.java

```diff
@@ -21,6 +21,10 @@
 
 public class ServerMacSignatureVerifierContext extends MacSignatureVerifierContext {
```

Comment:
How about using `ClientMacSignatureVerifierContext` that is used for keycloak to verify the data that a client creates and signs (e.g. request object, JWS client assertion for client authentication)?

`ServerMacSignatureVerifierContext` is used for keycloak to sign and verify JWS (e.g. access token, ID token). DPoP Proof is the data that a client creates and signs and keycloak verify.

As for DPoP, I'm afraid that HMAC method might not be used...